# MSc-Thesis-2021-22-Ketaki-H
## Supplementary file 1
#### Filename: matrix.tsv.gz
#### Contents: SNP calls for 3081 samples of An. gambiae, An. coluzzii, and An. arabiensis (dominant vectors of Africa), obtained from the Anopheles gambiae 1000 Genomes Project phase 3. (Raw data containing alleles per position for all samples, where 0 = allele absent, 1 = allele present)
## Supplementary file 2
#### Filename: full_metadata.tsv
#### Contents: Metadata for 3081 sample samples of An. gambiae, An. coluzzii, and An. arabiensis containing fields like sample_id, country, taxon, sex_call etc.
## Supplementary file 3
#### Filename: csv2itol.py
#### Contents: Python script to annotate phylogenetic tree by supplied dataset. Dataset highlighted on the tree using coloured ranges.   
## Supplementary file 4
#### Filename: csv2itol_text.py
#### Contents: Python script to annotate phylogenetic tree by supplied dataset. Dataset highlighted on the tree using text labels.   
## Supplementary file 5
#### Filename: genome_analysis.R
#### Contents: R script to perform comparative genome analysis by  (i) RSCU analysis, (ii) ENC analysis, and (iii) PCA by MDS
## Supplementary file 6  
#### Filename: Malaria_GeneTree.pdf
#### Contents: Phylogenetic gene trees for 13 mitochondrial genes of dominant malaria vectors. 
## Supplementary file 7
#### Filename: RSCU.xltx
#### Contents: RSCU estimates for each codon of the mitochondrial genome of dominant malaria vectors. The RSCU values > 1.6 are highlighted in green (overrepresented codons), whereas RSCU values < 0.6 are highlighted in red (underrepresented codons).
## Supplementary file 8  
#### Filename: ENC_GC3.csv
#### Contents: ENC and GC3 values of mitochondrial genomes of all dominant malaria vectors  

#### Phylogenetic_tree_genus_Anopheles : High quality image of phylogenetic tree of mitochondrial genomes of 89 Anopheles species added for reference
